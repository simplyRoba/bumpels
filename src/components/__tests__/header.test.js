import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import { StaticQuery } from "gatsby" // mocked

import Header from "../header"

const testTitle = "TestTitle"

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: testTitle,
        },
      },
    })
  )
})

describe("Header", () => {
  it("renders correctly", () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})
