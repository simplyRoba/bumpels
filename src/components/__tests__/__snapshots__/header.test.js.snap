// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light bg-white fixed-top"
    >
      <div
        class="container"
      >
        <div
          class="navbar-brand"
        >
          <a
            aria-label="logo"
            href="/"
          >
            <img
              alt="TestTitle"
              class="logo"
              src="/img/logo.svg"
            />
          </a>
        </div>
        <button
          aria-controls="navbarNav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="justify-content-lg-end navbar-collapse collapse"
          id="navbarNav"
        >
          <div
            class="navSiblingFade navbar-nav"
          >
            <a
              class="navSibling navLink"
              href="/blog"
            >
              Blog
            </a>
            <a
              class="navSibling navLink"
              href="/gallery"
            >
              Gallery
            </a>
            <a
              class="navSibling navLink"
              href="/merchandise"
            >
              Merchandise
            </a>
            <a
              class="navSibling navLink"
              href="/about"
            >
              About
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
